### PR TITLE
TSG: Why is this API Failing?

### PR DESCRIPTION
**Pull request by nibanks**
Adds a trouble shooting guide section for diagnosing why a particular API is failing. This recently came up with a .NET issue so I figured it'd be a prime candidate for documentation.

### CODE REVIEW

File: docs/TSG.md [hunk 1]
```diff
@@ -17,22 +17,19 @@ This document is meant to be a step-by-step guide for trouble shooting any issue
...
-#ifndef ERROR_QUIC_HANDSHAKE_FAILURE
-#define ERROR_QUIC_HANDSHAKE_FAILURE    _HRESULT_TYPEDEF_(0x80410000L)
-#endif
-
-#ifndef ERROR_QUIC_VER_NEG_FAILURE
-#define ERROR_QUIC_VER_NEG_FAILURE      _HRESULT_TYPEDEF_(0x80410001L)
-#endif
-
-...
+#define QUIC_STATUS_ADDRESS_IN_USE          HRESULT_FROM_WIN32(WSAEADDRINUSE)               // 0x80072740
```
Comment (davinci26) [inline docs/TSG.md:28]: nit: do you want to add a link to [QUIC_STATUS.md#well-known-status-codes](https://github.com/microsoft/msquic/blob/fa4e0d4375dc4573ce718f829ac1713622bdcda8/docs/api/QUIC_STATUS.md#well-known-status-codes) so it is easy to see the error description of each message

**Review by davinci26 — APPROVED** [review]
It looks clean to me and easy to follow

File: docs/TSG.md (hunk 1 above)
Comment (nibanks) [reply docs/TSG.md:28]: Thanks for the suggestion! Will do.

File: docs/TSG.md [hunk 2]
```diff
@@ -76,6 +75,44 @@ The error code indicated in this event is completely application defined (type o
...
+### Example (ListenerStart failing with QUIC_STATUS_INVALID_STATE)
+
+In a recent example, we wanted to know why an app occasionally received `QUIC_STATUS_INVALID_STATE` when it called `ListenerStart`. We took the following steps to diagnose it.
+
+1. Collected traces for the repro.
+2. Converted to text and opened them in [TextAnalysisTool](./Diagnostics.md#text-analysis-tool).
+3. Added a filter for all API enter events for `QUIC_TRACE_API_LISTENER_START` (`10`).
+4. Looked for the following `[ api] Exit` event after each enter event on the same `[process.thread]`.
+
+This quicky resulted in the following pair of events. They show the app called `ListenerStart` for the listener pointer `7f30ac0dcff0` at `09:54:03.528362` in process `2e73` on thread `2e8b` (CPU 1). Shortly after, MsQuic returned with status `200000002` (`QUIC_STATUS_INVALID_STATE` on Posix platforms).
```
Comment (maolson-msft) [inline docs/TSG.md:98]:
```suggestion
This quickly resulted in the following pair of events. They show the app called `ListenerStart` for the listener pointer `7f30ac0dcff0` at `09:54:03.528362` in process `2e73` on thread `2e8b` (CPU 1). Shortly after, MsQuic returned with status `200000002` (`QUIC_STATUS_INVALID_STATE` on Posix platforms).
```

**Review by maolson-msft — APPROVED** [review]

**Pull request merged**